src/ideascale/fetch.rs

```diff
@@ -0,0 +1,113 @@
+use crate::ideascale::models::de::{Fund, Funnel, Proposal};
+
+use once_cell::sync::Lazy;
+use serde::de::DeserializeOwned;
+use serde::Deserialize;
+use url::Url;
+
+use std::collections::HashMap;
+use std::convert::TryInto;
+
+#[derive(thiserror::Error, Debug)]
+pub enum Error {
+    #[error(transparent)]
+    RequestError(#[from] reqwest::Error),
+
+    #[error("Could not get value from json, missing attribute {attribute_name}")]
+    MissingAttribute { attribute_name: &'static str },
+}
+
+#[derive(Debug, Deserialize)]
+struct Score {
+    #[serde(alias = "ideaId")]
+    id: i32,
+    #[serde(alias = "avgScoreOfIdea")]
+    score: f32,
+}
+
+pub type Scores = HashMap<i32, f32>;
+
+static BASE_IDEASCALE_URL: Lazy<url::Url> = Lazy::new(|| {
+    "https://cardano.ideascale.com/a/rest/v1/"
+        .try_into()
+        .unwrap()
+});
+
+static CLIENT: Lazy<reqwest::Client> = Lazy::new(reqwest::Client::new);
```

Comment:
I'd rather use an explicit state object to instantiate the client with the API token, and make those fetch functions its methods.
Should not make much difference for the CLI tool, though.

Reply:
Yes, but it is messy to spawn clients in the async tasks later. It will avoid the api_token spreading though.

Reply:
API token retrieval from an environment variable could go under the lazy initialization, but that means you can't easily have the command line option for it (which the tool probably shouldn't, as per prior discussion)